### PR TITLE
[LogicApps] New Logic App SDK version

### PR DESCRIPTION
**Pull request by refortie**
## Description

Updated the Logic App SDK to the newest version. The main change of which was adding in Batch Configurations and Assemblies. It also added in some new Map Types and a B2B type.

This includes the fix for #7759, #6261, #2539 and #5233

[Completed Cmdlet Review](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/205)

This has been broken into several commits to make it easier to review as there are a lot of small changes to make the code compile

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/LogicApp/LogicApp.Test/ScenarioTests/IntegrationAccountAssemblyTests.ps1 [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+ <#
+.SYNOPSIS
+Test New-AzIntegrationAccountAssembly command
+#>
+function Test-CreateIntegrationAccountAssembly
+{
+	$localAssemblyFilePath = Join-Path $TestOutputRoot "\Resources\SampleAssembly.dll"
```
[the remark this answers is not in the document]
Comment (refortie) [reply src/LogicApp/LogicApp.Test/ScenarioTests/IntegrationAccountAssemblyTests.ps1]: I'm not sure what the action item is here. I believe I am using the "Join-Path" everywhere new, following the pattern that was added recently (October, I think, I'm having trouble finding it due to the mass rename) which is as above `Join-Path $TestOutputRoot "\Resources\SampleAssembly.dll"`. Should it be more like `Join-Path $TestOutputRoot "Resources" "SampleAssembly.dll"`?

[the remark this answers is not in the document]
Comment (maddieclayton) [reply src/LogicApp/LogicApp.Test/ScenarioTests/IntegrationAccountAssemblyTests.ps1]: Yes, that's correct - all parts of the path should be combined together using Join-Path

[the remark this answers is not in the document]
Comment (refortie) [reply src/LogicApp/LogicApp.Test/ScenarioTests/IntegrationAccountAssemblyTests.ps1]: Sounds good, changed everywhere

**Comment by refortie** [comment]
Rebased and force pushed to fix merge conflicts

**Comment by refortie** [comment]
@maddieclayton Could you advise how to solve the static analysis issue, the documentation linked doesn't talk about the SharedAssemblyConflict.csv file

**Comment by maddieclayton** [comment]
The build is failing because you are taking a dependency on Newtonsoft Json 11.  We already have a reference to Newtonsoft 10.0.3 for all our modules, so this new dependency can't be added to your module.  Is there something from 11 that is not available in 10?

**Comment by refortie** [comment]
I think when I originally started this, it was locked to newtonsoft 6 or something. I've pulled out the override and pushed

**Comment by maddieclayton** [comment]
Please fix credscan issues: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell/10256/artifact/src/Package/CredentialScannerOutput/CredScanIssues.csv/*view*/
Instructions:
```
16:56:29 For false positives, follow the instructions on this doc: https://github.com/Azure/adx-documentation-pr/blob/master/engineering/credscan-ps.md
16:56:29 For more information, please read the 'How should I respond to reports of incorrectly handled secrets?' section 
16:56:29 of this doc: https://microsoft.sharepoint.com/teams/AzureSecurityCompliance/Security/SitePages/SecretsManagement.aspx.
```

**Comment by refortie** [comment]
@MiYanni Is there a particular meaning to the Do Not Merge label that I should worry about?

@maddieclayton If you could take another look, that would be great

**Comment by MiYanni** [comment]
@refortie Last time there was a PR like this (multiple issue fixes, design review additions/changes, and SDK update), it took a few weeks to go through with that service team. If there is any way that you can make the changes so there is a PR to encapsulate each of the logical changes, that would make it a lot easier to review and merge progressively. Almost 300 changes is a lot to handle at once, and things can be accidentally overlooked.

**Comment by refortie** [comment]
@MiYanni Just want to point out that over half the files are SessionRecords and regenerated help files. The most important commit to look at is the first one, I specifically separated those out to make it easier to review. Of the remaining 80ish files, the vast majority of those are just adding in the required "this" prefix.

I'll try to break it down more, but the first PR is still going to have 150 files in it, since I still have to make the mandatory changes in one merge

**Comment by MiYanni** [comment]
@refortie `this` is not required. Why do you think it is required? Here are the correct uses of `this`: https://stackoverflow.com/a/23502/294804
There are naming conventions in C# where private fields use `_` prefix as to not conflict with variable/parameter names. For example, `_logger` used in tests. You never, for any reason, need to access `_logger` as `this._logger` as long as the naming convention is followed. These are C# standards we (Microsoft) have created. Static analyzers (like Resharper) tell you to remove redundant `this.` usage. So, I delete unnecessary `this.` usage whenever I get the chance.

**Comment by refortie** [comment]
I agree with you about them and wanting to remove them. I'll be honest I thought it was a MS standard, but I'm guessing it was a (my) team level one, that required them. I'll manually remove the main ones that I add. If I rely on VS to do it, it'll probably just create the problem in the other direction

**Comment by MiYanni** [comment]
@refortie Understood. It is just a big pet-peeve of mine to use `this.` unnecessarily. It adds no value and hinders readability. Here is a nice cheat sheet of naming conventions: https://github.com/ktaranov/naming-convention/blob/master/C%23%20Coding%20Standards%20and%20Naming%20Conventions.md This is followed to avoid name clashes (primary use of `this.`). 

We do, however, have an extension method that requires the use of `this.`, `this.IsBound()`. We created it to make it easier to tell if PowerShell parameters have been bound.

**Comment by maddieclayton** [comment]
Please open this as two to three PRs - one that just bumps the SDK, one that introduces the new cmdlets in the review referenced, and one that fixes the bugs (if they are separate from simply updating the SDK).

As is, this PR is unmanageably large and I don't want to miss big issues or possible breaking changes.

**Pull request closed without merging**